### DISCUSSION TITLE
Encrypted secrets and Yarn 2+

### DISCUSSION THREAD

**Discussion by colbymelvin** [renovatebot/renovate#20331] [Request Help] [answered]
### I'm using the Mend Renovate hosted app.

- [X] I'm using the Mend Renovate hosted app.

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe your problem.

Hi there!

I'm currently setting up Renovate in a private repo for our org. We're using Yarn 2+ with PnP and "zero intalls", and we have defined `npmScopes` to handle authentication for a private registry.

I'm reading through the [documentation](https://docs.renovatebot.com/getting-started/private-packages/#encrypting-secrets) describing how to setup hostRules with encrypted secrets and it seems pretty straightforward.

However, your [documentation](https://docs.renovatebot.com/getting-started/private-packages/#yarn-2) on Yarn 2+ states the following:

> Renovate updates npmRegistries in .yarnrc.yml with resolved hostRules before running Yarn. For Renovate to overwrite existing npmRegistries entry, ...

It the code sample, it looks like Renovate will persist the _decrypted_ secret to `.yarnrc.yml`. That seem a little concerning to me - it also directly contradicts the docs on how/why to setup encrypted secrets, specifically:

> It is strongly recommended that you avoid committing secrets to repositories, including private ones, and this includes secrets needed by Renovate to access private modules.

We don't want to introduce this decrypted secret into our repository. So, a couple questions:

1. does Renovate actually commit/persist this change to `.yarnrc.yml`?
2. if so, does it need to? can we bypass that such that Renovate only has the decrypted secret for the lifecycle of the Renovate job run?

### Relevant debug logs.

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
renovate doesn't commit that file, it should normally deleted/ reset after renovate finished the branch locally.

it's just temporary.

**Reply by colbymelvin** [reply]
Thanks for the clarification!
